specs/phase1/beacon-chain.md

```diff
@@ -1057,51 +485,124 @@ def process_epoch(state: BeaconState) -> None:
+    shards = [
+        compute_shard_from_committee_index(state, index, slot)
+        for i in range()
+        state,
+        attestation.data.index,
+        attestation.data.slot
+    )
+    state.current_epoch_pending_headers = []
+    # Add dummy "empty" PendingAttestations
+    # (default to vote for if no shard header available)
+    for slot in range(SLOTS_IN_EPOCH):
+        for index in range(get_committee_count_per_slot(get_current_epoch(state))):
+            shard = compute_shard_from_committee_index(state, index, slot)
+            committee_length = len(get_beacon_committee(
+                state,
+                header.slot,
+                header.shard
+            ))
+            state.current_epoch_pending_headers.append(PendingShardHeader(
+                slot=slot,
+                shard=shard,
+                commitment=DataCommitment(),
+                root=Root(),
+                votes=Bitlist[MAX_COMMITTEE_SIZE]([0] * committee_length),
+                confirmed=False
+            ))
+
+```
+
 #### Phase 1 final updates
 
 ```python
 def process_phase_1_final_updates(state: BeaconState) -> None:
     process_custody_final_updates(state)
-    process_online_tracking(state)
-    process_light_client_committee_updates(state)
-
     # Update current_epoch_start_shard
     state.current_epoch_start_shard = get_start_shard(state, Slot(state.slot + 1))
 ```
 
 #### Custody game updates
 
-`process_reveal_deadlines`, `process_challenge_deadlines` and `process_custody_final_updates` are defined in [the Custody Game spec](./custody-game.md), 
-
-#### Online-tracking
-
-```python
-def process_online_tracking(state: BeaconState) -> None:
-    # Slowly remove validators from the "online" set if they do not show up
-    for index in range(len(state.validators)):
-        if state.online_countdown[index] != 0:
-            state.online_countdown[index] = state.online_countdown[index] - 1
-
-    # Process pending attestations
-    for pending_attestation in state.current_epoch_attestations + state.previous_epoch_attestations:
-        for index in get_attesting_indices(state, pending_attestation.data, pending_attestation.aggregation_bits):
-            state.online_countdown[index] = ONLINE_PERIOD
-```
-
-#### Light client committee updates
-
-```python
-def process_light_client_committee_updates(state: BeaconState) -> None:
-    """
-    Update light client committees.
-    """
-    next_epoch = compute_epoch_at_slot(Slot(state.slot + 1))
-    if next_epoch % LIGHT_CLIENT_COMMITTEE_PERIOD == 0:
-        state.current_light_committee = state.next_light_committee
-        new_committee = get_light_client_committee(state, next_epoch + LIGHT_CLIENT_COMMITTEE_PERIOD)
-        state.next_light_committee = committee_to_compact_committee(state, new_committee)
-```
+`process_reveal_deadlines`, `process_challenge_deadlines` and `process_custody_final_updates` are defined in [the Custody Game spec](./custody-game.md).
```

Comment:
Note that some custody game SSZ fields are removed in this PR.

Do you intend to split the custody game into another fork? Or it's a temporary removal?

Reply:
I'm inclined to say, first stabilize this document somewhat without caring too much about custody, and then after a few weeks start figuring out how to bring the custody games back?